File: Session_1/Bài 1_FindMax/1_FindMax.cpp
#include <iostream> 
#include <vector>    
#include <limits.h>  
using namespace std;

// Hàm FindMax: Tìm giá trị lớn nhất trong một mảng (vector) các số nguyên
int FindMax(const vector<int>& A) {
    // Kiểm tra nếu mảng rỗng
    if (A.empty()) {
        cerr << "Error: Empty array!" << endl; // Thông báo lỗi
        return INT_MIN; // Trả về giá trị nhỏ nhất của kiểu int để báo lỗi
    }

    //  phần tử đầu tiên là giá trị lớn nhất
    int max = A[0];

    // Lặp qua các phần tử còn lại của mảng để tìm giá trị lớn nhất
    for (size_t i = 1; i < A.size(); ++i) {
        if (A[i] > max) { //  phần tử hiện tại lớn hơn giá trị lớn nhất hiện tại
            max = A[i];   // Cập nhật giá trị lớn nhất
        }
    }

    return max; // Trả về giá trị lớn nhất
}

int main() {
    // Khởi tạo một mảng (vector) chứa các số nguyên
    vector<int> A = {3, 5, 2, 9, 1};

    // Gọi hàm FindMax để tìm giá trị lớn nhất trong mảng
    int maxVal = FindMax(A);

    // Nếu giá trị trả về không phải là INT_MIN, in ra kết quả
    if (maxVal != INT_MIN) {
        cout << "The largest number in the sequence is " << maxVal << endl;
    }

    return 0;
}

File: Session_1/Bài 1_FindMax/1_FindMax.py
# Hàm FindMax: Tìm giá trị lớn nhất trong một danh sách (list)
def FindMax(A):
    # Kiểm tra nếu danh sách rỗng
    if not A:
        print("Error: Empty array!")  # Thông báo lỗi
        return None  # Trả về None để biểu thị lỗi
    
    # Giả sử phần tử đầu tiên là giá trị lớn nhất
    max_val = A[0]

    # Lặp qua các phần tử còn lại để tìm giá trị lớn nhất
    for i in range(1, len(A)):
        if A[i] > max_val:  # Nếu phần tử hiện tại lớn hơn giá trị lớn nhất
            max_val = A[i]  # Cập nhật giá trị lớn nhất
    
    return max_val  # Trả về giá trị lớn nhất

# Hàm main để khởi chạy chương trình
def main():
    # Khởi tạo danh sách các số nguyên
    A = [3, 5, 2, 9, 1]
    
    # Gọi hàm FindMax để tìm giá trị lớn nhất trong danh sách
    max_val = FindMax(A)
    
    # Nếu kết quả không phải None, in ra giá trị lớn nhất
    if max_val is not None:
        print(f"Giá trị lớn nhất trong mảng là: {max_val}")

# Điểm khởi đầu của chương trình
if __name__ == "__main__":
    main()
